components/Ui/Link/UiLink.vue

```diff
@@ -1,60 +1,83 @@
 <template>
-  <component
-    :is="isNuxtLink ? 'NuxtLink' : 'a'"
+  <NuxtLink
     v-track-click="segment"
-    :href="isAnchor && url"
-    :to="isNuxtLink && url"
-    :style="hasLink && 'cursor:pointer'"
-    :rel="isExternal && 'noopener'"
-    :target="isExternal ? '_blank' : undefined"
+    :external="computedExternal"
+    :target="computedTarget"
     :title="title"
+    :to="url"
     @click="emit('click')"
     @mouseenter="$emit('mouseenter')"
   >
     <slot />
-  </component>
+  </NuxtLink>
 </template>
 
 <script setup lang="ts">
 import { CtaClickedEventProp } from "~/types/segment";
 
+type NuxtLinkTarget =
+  | "_blank"
+  | "_parent"
+  | "_self"
+  | "_top"
+  | (string & {})
+  | null;
+
 interface Props {
+  external?: boolean;
```

Comment:
As far as I know optional parameters in Typescript automatically _translate_ to "or `undefined`". (https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties)

But this made me realise that I was adding unnecessary `undefined` default values to the props, which can be safely removed. I did that here: https://github.com/Qiskit/qiskit.org/pull/3436/commits/e6d3fcae2d79f096e486d4d95b0263eefb0b67a1

Reply:
As discussed offline, regarding having this prop have `false` as default instead of `undefined`:

> In this case I think that it makes sense to leave it with `undefined` because we might actually not define the value when we pass these props. Also, the check that is done to the `external` prop is to see if it is not _falsey_, which works fine as-is. Additionally, this is closer the NuxtLink's implementation.
> 
> In summary, I think that this is "more correct", having it default to `undefined` in this specific case.